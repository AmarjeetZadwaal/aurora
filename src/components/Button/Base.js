import React from "react";
import PropTypes from "prop-types";

import { StyledButton, StyledButtonLink } from "./Base.styles";
import BUTTON_VARIANTS from "./constants";
import getRelByTarget from "../../utils/link";

const Button = ({ variant, children, ...rest }) => {
  const { href } = rest;

  if (href) {
    const { rel, target } = rest;
    const validatedRel = getRelByTarget(target, rel);

    return (
      <StyledButtonLink variant={variant} rel={validatedRel} {...rest}>
        {children}
      </StyledButtonLink>
    );
  }

  return (
    <StyledButton variant={variant} {...rest}>
      {children}
    </StyledButton>
  );
};

Button.propTypes = {
  variant: PropTypes.oneOf(BUTTON_VARIANTS).isRequired,
  children: PropTypes.node.isRequired
};

export default Button;
