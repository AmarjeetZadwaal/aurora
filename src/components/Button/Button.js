import React from "react";
import PropTypes from "prop-types";

import { StyledButton, StyledButtonLink } from "./Button.styles";
import BUTTON_VARIANTS from "./constants";

const Button = ({ variant, children, ...rest }) => {
  const { href, target } = rest;

  if (href) {
    const rel = (target === "_blank" && "noopener") || undefined;

    return (
      <StyledButtonLink variant={variant} rel={rel} {...rest}>
        {children}
      </StyledButtonLink>
    );
  }

  return (
    <StyledButton variant={variant} {...rest}>
      {children}
    </StyledButton>
  );
};

Button.propTypes = {
  variant: PropTypes.oneOf(BUTTON_VARIANTS).isRequired,
  children: PropTypes.node.isRequired
};

export default Button;
