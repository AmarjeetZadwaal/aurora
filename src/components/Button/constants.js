const BUTTON_VARIANTS = ["standard", "outline", "transparent"];

export default BUTTON_VARIANTS;
