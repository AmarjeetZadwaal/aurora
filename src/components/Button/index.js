import React from "react";

import { StyledButton, StyledButtonLink } from "./index.styles";

const Button = ({ variant, children, ...rest }) => {
  const Btn = rest.href ? StyledButtonLink : StyledButton;
  return (
    <Btn variant={variant} {...rest}>
      {children}
    </Btn>
  );
};

export default Button;
