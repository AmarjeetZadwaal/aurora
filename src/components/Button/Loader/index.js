import React from "react";
import PropTypes from "prop-types";

import { Dot, LoaderWrapper, ContentWrapper, DotsWrapper } from "./styles";
import { BUTTON_VARIANTS } from "../../constants";

const Loader = ({ children, isLoading, variant }) => (
  <LoaderWrapper>
    {isLoading && (
      <DotsWrapper data-bdd="loader">
        <Dot variant={variant} />
        <Dot variant={variant} />
        <Dot variant={variant} />
      </DotsWrapper>
    )}
    <ContentWrapper isLoading={isLoading}>{children}</ContentWrapper>
  </LoaderWrapper>
);

Loader.propTypes = {
  isLoading: PropTypes.bool,
  variant: PropTypes.oneOf(BUTTON_VARIANTS).isRequired,
  children: PropTypes.node.isRequired
};

Loader.defaultProps = {
  isLoading: false
};

export default Loader;
