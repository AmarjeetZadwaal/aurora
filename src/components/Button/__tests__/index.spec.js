import React from "react";
import renderer from "react-test-renderer";

import Button from "../";

describe("<Button />", () => {
  it("renders standard button correctly", () => {
    const component = renderer.create(
      <Button variant="standard">Dummy Label</Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders standard disabled button correctly", () => {
    const component = renderer.create(
      <Button variant="standard" disabled>
        Dummy Label
      </Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders outline button correctly", () => {
    const component = renderer.create(
      <Button variant="outline">Dummy Label</Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders outline disabled button correctly", () => {
    const component = renderer.create(
      <Button variant="outline" disabled>
        Dummy Label
      </Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders link button correctly", () => {
    const component = renderer.create(
      <Button href="/" variant="standard">
        Dummy Label
      </Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
