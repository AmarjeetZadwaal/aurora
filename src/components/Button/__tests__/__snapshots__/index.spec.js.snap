// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders link button correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  font-weight: 600;
  padding: 0 14px;
  height: 36px;
  width: 100%;
  min-width: 100px;
  text-align: center;
  text-transform: capitalize;
  border-radius: 2px;
  margin-right: 14px;
  cursor: pointer;
  color: #fff;
  background-color: #009cde;
  border: 1px solid #009cde;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:disabled {
  color: #fff;
  background-color: #bfe6f7;
  border: 1px solid #bfe6f7;
  cursor: not-allowed;
}

<a
  className="c0"
  href="/"
>
  Dummy Label
</a>
`;

exports[`<Button /> renders outline button correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  font-weight: 600;
  padding: 0 14px;
  height: 36px;
  width: 100%;
  min-width: 100px;
  text-align: center;
  text-transform: capitalize;
  border-radius: 2px;
  margin-right: 14px;
  cursor: pointer;
  color: #009cde;
  background-color: #fff;
  border: 1px solid #009cde;
}

.c0:disabled {
  color: #bfe6f7;
  background-color: #fff;
  border: 1px solid #bfe6f7;
  cursor: not-allowed;
}

<button
  className="c0"
>
  Dummy Label
</button>
`;

exports[`<Button /> renders outline disabled button correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  font-weight: 600;
  padding: 0 14px;
  height: 36px;
  width: 100%;
  min-width: 100px;
  text-align: center;
  text-transform: capitalize;
  border-radius: 2px;
  margin-right: 14px;
  cursor: pointer;
  color: #009cde;
  background-color: #fff;
  border: 1px solid #009cde;
}

.c0:disabled {
  color: #bfe6f7;
  background-color: #fff;
  border: 1px solid #bfe6f7;
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
>
  Dummy Label
</button>
`;

exports[`<Button /> renders standard button correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  font-weight: 600;
  padding: 0 14px;
  height: 36px;
  width: 100%;
  min-width: 100px;
  text-align: center;
  text-transform: capitalize;
  border-radius: 2px;
  margin-right: 14px;
  cursor: pointer;
  color: #fff;
  background-color: #009cde;
  border: 1px solid #009cde;
}

.c0:disabled {
  color: #fff;
  background-color: #bfe6f7;
  border: 1px solid #bfe6f7;
  cursor: not-allowed;
}

<button
  className="c0"
>
  Dummy Label
</button>
`;

exports[`<Button /> renders standard disabled button correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  font-weight: 600;
  padding: 0 14px;
  height: 36px;
  width: 100%;
  min-width: 100px;
  text-align: center;
  text-transform: capitalize;
  border-radius: 2px;
  margin-right: 14px;
  cursor: pointer;
  color: #fff;
  background-color: #009cde;
  border: 1px solid #009cde;
}

.c0:disabled {
  color: #fff;
  background-color: #bfe6f7;
  border: 1px solid #bfe6f7;
  cursor: not-allowed;
}

<button
  className="c0"
  disabled={true}
>
  Dummy Label
</button>
`;
